### PR TITLE
handle 404s for series and story 

### PR DESCRIPTION
**Pull request by gcampo88**
#407

### CODE REVIEW

File: src/app/series/series.component.ts [hunk 1]
```diff
@@ -37,11 +37,19 @@ export class SeriesComponent implements OnInit {
...
-        });
-      });
+      this.cms.auth.follow('prx:series', {id: this.id}).subscribe(
+        s => {
+          s.follow('prx:account').subscribe(a => {
+            this.setSeries(a, s);
+          });
+        },
+        err => {
+          if (err.status === 404 && err.name === 'HalHttpError') {
```
Comment (cavis) [inline src/app/series/series.component.ts]: Does typescript not freak out about "unknown property: status" here?

Comment (gcampo88) [inline src/app/series/series.component.ts]: nope! I wonder why...

File: src/app/core/cms/index.ts [hunk 2]
```diff
@@ -4,4 +4,4 @@ export { HalDoc } from './haldoc';
 export { HalLink } from './hallink';
 export { HalObservable } from './halobservable';
 export { HalRemoteCache } from './halremote.cache';
-export { HalRemote } from './halremote';
+export { HalRemote, HalHttpError } from './halremote';
```
Comment (cavis) [inline src/app/core/cms/index.ts]: Note to self: PR #413 is going to have fun merging this.

File: src/app/series/series.component.ts [hunk 3]
```diff
@@ -37,11 +37,20 @@ export class SeriesComponent implements OnInit {
...
```
Comment (cavis) [inline src/app/series/series.component.ts]: I'm not sure checking `err.name` is the usual typescript way of doing this - does `err instanceof HalHttpError` work?  (Not a huge deal, though).

File: src/app/series/series.component.ts [hunk 4]
```diff
@@ -37,11 +37,20 @@ export class SeriesComponent implements OnInit {
...
+            this.toastr.error('No series found. Redirecting to new series page');
+            console.error(`Series with id ${this.id} not found`);
+            setTimeout(() => this.router.navigate(['/series', 'new']), 3000);
+          }
```
Comment (cavis) [inline src/app/series/series.component.ts]: If you didn't handle the error here, you should probably re-throw it so it doesn't just disappear.

File: src/app/story/story.component.ts [hunk 5]
```diff
@@ -60,13 +60,21 @@ export class StoryComponent implements OnInit {
...
+            story.follow('prx:series').subscribe(series => this.setStory(series, story));
+          } else {
+            this.setStory(null, story);
+          }
+        },
+        err => {
+          if (err.status === 404 && err.name === 'HalHttpError') {
+            this.toastr.error('No episode found. Redirecting to new episode page');
+            setTimeout(() => this.router.navigate(['/story', 'new']), 3000);
+          }
```
Comment (cavis) [inline src/app/story/story.component.ts]: Again, re-throw unhandled errors.

File: src/app/core/cms/index.ts (hunk 2 above)
Comment (gcampo88) [inline src/app/core/cms/index.ts]: :)

File: src/app/series/series.component.ts (hunk 3 above)
Comment (gcampo88) [inline src/app/series/series.component.ts]: `err instanceof HalHttpError` does not work -- not sure why! But looks like we've run into that before --
 https://github.com/PRX/publish.prx.org/blob/master/src/app/story/directives/metrics-downloads.component.ts#L85

File: src/app/series/series.component.ts (hunk 4 above)
Comment (gcampo88) [inline src/app/series/series.component.ts]: 👍

File: src/app/story/story.component.ts (hunk 5 above)
Comment (gcampo88) [inline src/app/story/story.component.ts]: 👍

File: src/app/series/series.component.ts (hunk 3 above)
Comment (cavis) [inline src/app/series/series.component.ts]: May be because extending `Error` in typescript is a bit wonky in the first place.  Ah well... this works!

**Pull request merged**